converter_app/readers/stoe_raw.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+from converter_app.readers.helper.reader import Readers
+from converter_app.readers.ascii import AsciiReader
+
+logger = logging.getLogger(__name__)
+
+import struct
+import numpy as np
+
+
+
+class StoeRawReader(AsciiReader):
+    """
+    Implementation of the TVB Reader for reading Stoe raw binary data file.
+    Code was extracted from the TVB jupiter Notebook by Ron Dockhorn.
+    # Beginn of Notebook
+    # Simple Conversion of Stoe raw data file
+    # by Ron Dockhorn (https://orcid.org/0000-0002-5268-5430)
+    """
+    identifier = 'stoe_raw_reader'
+    priority = 5
+
+    contentrawfile = ''
+    file_type = ''
+
+    def unpack_repeated_bytes(self, byte_data, data_type, count):
+        """
+        Unpack a series of bytes into a tuple of the same data type.
+
+        :param byte_data: The bytes to unpack.
+        :param data_type: The format character for the data type (e.g., 'b' for signed char).
+        :param count: The number of items to unpack.
+        :return: A tuple of unpacked values.
+        """
+        # Create the format string based on the data type and count
+        # 'i': Integer (4 bytes)
+        # 'f': Float (4 bytes)
+        # 'd': Double (8 bytes)
+        # 'h': Short (2 bytes)
+        # 'b': Signed char (1 byte)
+        # 'B': Unsigned char (1 byte)
+        # 'q': Long long (8 bytes)
+        # 'Q': Unsigned long long (8 bytes)
+        format_string = f'{count}{data_type}'
+
+        # Unpack the byte data using the constructed format string
+        return struct.unpack(format_string, byte_data)
+
+    def get_non_empty_chunks_separated_by_null(self, data_slice):
+        """
+        Get all non-empty chunks of data separated by NULL bytes.
+
+        :param data_slice: The slice of data to split.
+        :return: A list of bytes objects, each representing a non-empty chunk of data.
+        """
+        # Split the data by NULL bytes and filter out empty chunks
+        return [chunk for chunk in data_slice.split(b'\x00') if chunk]
+
+    def get_file_type_version(self):
+        ###
+        # File Type Version
+        ###
+
+        self.contentrawfile = self.file.content
+
+        count = len(self.contentrawfile[0x00:0x0D + 1]) // 1  # Number of bytes to unpack
+
+        unpacked_data = self.unpack_repeated_bytes(self.contentrawfile[0x00:0x0D + 1], 'b', count)
+
+        # Convert unpacked data to a string
+        string_output_file_type = ''.join(chr(b) for b in unpacked_data)
+
+        return string_output_file_type
+
+    def check(self):
+        """
+        :return: True if it fits
+        """
+
+        return (
+                self.file.suffix.lower() == ".raw"
+                and self.file.encoding.lower() == "binary"
+                and self._cache_file_type() == "RAW_1.06Powdat"
+        )
+
+    def _cache_file_type(self) -> str:
```

Comment:
I think it would be nicer as Property

```python
@property
def file_type(self):
     if self._file_type is None:
          self._file_type == ....
     return self._file_type
```